infra/helper.py

```diff
@@ -309,6 +309,10 @@ def _workdir_from_dockerfile(project_name):
       workdir = match.group(1)
       workdir = workdir.replace('$SRC', '/src')
 
+      # To prevent overwriting /src with contents of a local checkout.
+      if workdir.rstrip(os.path.sep) == '/src':
```

Comment:
Why do we worry about such cases? My understanding is that if we mount /src we no longer have the build.sh/etc files but I think such builds with local checkouts would be broken in these cases anyway since the paths are wrong?

Reply:
Great catch! This indeed doesn't solve the problem properly. Uploading an update.